crates/core/src/config.rs

```diff
@@ -84,25 +84,37 @@ impl Default for Config {
 }
 
 impl Config {
-    pub fn from_env() -> Self {
-        Self {
-            network: Network::from_str(
-                &std::env::var("NETWORK").unwrap_or_default(),
-            )
-            .unwrap_or(Network::MAINNET),
-            eth_execution_rpc: std::env::var("ETH_EXECUTION_RPC")
-                .unwrap_or_default(),
-            starknet_rpc: std::env::var("STARKNET_RPC").unwrap_or_default(),
-            data_dir: PathBuf::from(
-                std::env::var("DATA_DIR").unwrap_or_default(),
-            ),
-            poll_secs: u64::from_str(
-                &std::env::var("POLL_SECS").unwrap_or_default(),
-            )
-            .unwrap_or(DEFAULT_POLL_SECS),
-            rpc_addr: rpc_addr(),
-            fee_token_addr: fee_token_addr(),
-        }
+    pub fn from_env() -> Result<Self> {
+        Self::from_vars(|key| std::env::var(key).ok())
+    }
+
+    fn from_vars<F>(get: F) -> Result<Self>
+    where
+        F: Fn(&'static str) -> Option<String> + 'static,
+    {
+        let require = |var_key: &'static str| {
+            get(var_key).ok_or_else(|| eyre!("The \"{}\" env var must be set or a configuration file must be specified", var_key))
+        };
+
+        Ok(Self {
+            network: Network::from_str(&get("NETWORK").unwrap_or_default())
```

Comment:
if the provided network is invalid, then mainnet is used as a default which might not be always desired behavior; I think an error must be returned when provided network is invalid - with respective test case.